.github/workflows/ci.yml

```diff
@@ -0,0 +1,125 @@
+# This is a the GitHub Action to run the CI for SODA Dashboard
+
+name: SODA Installer CI
+
+# Controls when the action will run. 
+# Triggers the workflow on push or pull request events but only for the main branch
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        go-version: [1.13.x]
+        os: [ubuntu-16.04]
+        
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - name: Setup Go environment
+        uses: actions/setup-go@v2.1.3
+        with:
+          # The Go version to download (if necessary) and use. Supports semver spec and ranges.
+          go-version: ${{ matrix.go-version }}
+          # Whether to download only stable versions
+          stable: true # optional, default is true
+          
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: checkout the repository
+        uses: actions/checkout@v2
+      
+      - name: List ansible sources in the runtime
+        run: |
+            echo "Installed Ansible Version in Runtime: "
+            ansible --version
+        
+      - name: List ansible sources and  Remove latest ansible on Ubuntu 16.04
+        run: |
+            echo "List the ansible sources"
+            ls /etc/apt/sources.list.d | grep ansible
+            echo "Remove ansible sources other than the required version"
+            sudo rm /etc/apt/sources.list.d/ansible-ubuntu-ansible-xenial.list /etc/apt/sources.list.d/ansible-ubuntu-ansible-xenial.list.save
+            echo "Updating the pacakage list"
+            sudo apt-get update
+        if: ${{ matrix.os == 'ubuntu-16.04' }}
+
+      # Change the permissions of the ansible installer
+      - name: 'Make ansible installer executable and run install ansible on OS: ${{ matrix.os }}'
+        run: |
+            sudo chmod +x ./ansible/install_ansible.sh
+            ./ansible/install_ansible.sh
+        shell: bash
+        env:
+          CI: true
+        if: ${{ always() }}
+
+      - name: Check Python and Python3 version
+        run: |
+            echo "Python version:"
+            python -V
+            echo "Python3 version:"
+            python3 -V
+        if: ${{ matrix.os == 'ubuntu-16.04' }}
+
+      - name: Setup and update Python3 to version 3.6 on Ubuntu 16.04
+        run: |
```

Comment:
Suggestion : Can we move this into script, and call that script in yaml?

Reply:
@asifdxtreme Yes eventually i will move this to a script or better yet if I can get an action to do this I will.

Reply:
Please open an issue

Reply:
> Please open an issue

This is already done. The steps are now part of a script that is invoked only for Ubuntu16.04